draft-ietf-quic-transport.md

```diff
@@ -3270,14 +3270,25 @@ delivery, as long as it is not in violation of the receiver's flow control
 limits.
 
 An endpoint MUST NOT send data on any stream without ensuring that it is within
-the data limits set by its peer.  The cryptographic handshake stream, Stream 0,
-is exempt from the connection-level data limits established by MAX_DATA. Data on
-stream 0 other than the initial cryptographic handshake message is still subject
-to stream-level data limits and MAX_STREAM_DATA. This message is exempt
-from flow control because it needs to be sent in a single packet regardless of
-the server's flow control state. This rule applies even for 0-RTT handshakes
-where the remembered value of MAX_STREAM_DATA would not permit sending a full
-initial cryptographic handshake message.
+the data limits set by its peer.
+
+The cryptographic handshake stream, Stream 0, is exempt from the
+connection-level data limits established by MAX_DATA.  That is, data sent on
+stream 0 is not counted against the limit expressed by MAX_DATA.  Data on stream
+0 is still subject to stream-level data limits and MAX_STREAM_DATA.
+
+Data sent in the Initial and Retry packets do not count toward stream 0 flow
+control limits.  These packet types do not permit the sending of data in
+multiple packets, so there is no opportunity for a peer to send MAX_STREAM_DATA
+frames.  Note also that the next packet sent after either of these packets might
+start at a stream offset of 0.  Thus, the only limit on the size of the
+cryptographic handshake messages these packets contain is determined by the MTU.
+
+Endpoints MUST respect the stream flow control limit set by their peer.  A
+client might exceed this limit when sending an Initial packet and so might need
+to await a MAX_STREAM_DATA frame from the server before it can sent additional
+STREAM frames.  A client MAY send ACK frames and STREAM_BLOCKED frames for
+stream 0.
```

Comment:
Given that ACK frames don't reference a stream, the idea of an ACK frame for stream 0 seems undefined.  If that phrase wasn't supposed to apply to the ACK frames as well, then I'm having trouble parsing it.

Reply:
These are two separate things:

* ACK frames
* STREAM_BLOCKED frames for stream 0.

Don't know how to make that clearer, since - as you say - your interpretation is nonsensical.

Reply:
The statement that "A client MAY send ACK frames" is undoubtedly true, but feels like a bit of a non sequitur without some additional connection to the rest of the paragraph.  Perhaps, "A client in this state SHOULD send a STREAM_BLOCKED frame for stream 0 in a Handshake packet.  This packet MAY also include ACK frames."

Reply:
Yeah, I removed ACK.